test/liquidityConversionRate.js

```diff
@@ -869,6 +879,211 @@ contract('kyberReserve for Liquidity', function(accounts) {
+                // buy then check sell rate
+                let lastBuyRate = BigNumber(precision).div(newP0s[id]).floor();
+                let srcAmount = maxCapBuyInWei;
+                while (true) {
+                    let bal = await token.balanceOf(reserveInst.address);
+                    if (bal * 1.0 <= precision * 1.0) { break; }
+                    let buyRate = await reserveInst.getConversionRate(ethAddress, token.address, srcAmount, currentBlock);
+                    if (buyRate.valueOf() == 0) {
+                        while (true) {
+                            srcAmount = srcAmount.div(2).floor();
+                            if (srcAmount.valueOf() == 0) { break; }
+                            buyRate = await reserveInst.getConversionRate(ethAddress, token.address, srcAmount, currentBlock);
+                            if (buyRate.valueOf() != 0) { break; }
+                        }
+                        if (buyRate.valueOf() == 0) { break; }
+                    }
+                    lastBuyRate = buyRate;
+
+                    await reserveInst.trade(ethAddress, srcAmount, token.address, user1, buyRate, true, {from:network, value: srcAmount});
+                }
+                let minBuyRate = BigNumber(precision).div(pMax).floor();
+                assert(lastBuyRate * 1.0 >= minBuyRate * 1.0, "rate must be greater than min rate, loop: " + i);
+            } else {
+                // sell then check buy rate
+                let lastSellRate = BigNumber(precision).mul(newP0s[id]).floor();
+                let srcAmount = maxCapSellInWei.div(newP0s[id]).floor();
+                while (true) {
+                    let ethBal = await Helper.getBalancePromise(reserveInst.address);
+                    if (ethBal * 1.0 < precision.div(100) * 1.0) { break; } // less than 0.001 ether
+                    let sellRate = await reserveInst.getConversionRate(token.address, ethAddress, srcAmount, currentBlock);
+                    if (sellRate.valueOf() == 0) {
+                        while (true) {
+                            srcAmount = srcAmount.div(2).floor();
+                            sellRate = await reserveInst.getConversionRate(token.address, ethAddress, srcAmount, currentBlock);
+                            if (sellRate.valueOf() != 0) { break; }
+                        }
+                        if (sellRate.valueOf() == 0) { break; }
+                    }
+                    lastSellRate = sellRate;
+
+                    await token.transfer(network, srcAmount.valueOf());
+                    await token.approve(reserveInst.address, srcAmount, {from: network});
+                    await reserveInst.trade(token.address, srcAmount, ethAddress, user1, sellRate, true, {from: network});
+                }
+                let minSellRate = BigNumber(precision).mul(pMin).floor();
+                assert(lastSellRate * 1.0 >= minSellRate * 1.0, "rate must be greater than min sell rate, loop: " + i);
```

Comment:
please log details to understand test outcomes

Reply:
Added logs